internal/metrics/recorder.go

```diff
@@ -241,15 +240,20 @@ func (r *Recorder) RecordAddonMetrics(addon *addonsv1alpha1.Addon) {
 }
 
 func (r *Recorder) recordAddonHealthInfo(addon *addonsv1alpha1.Addon) {
+	// cleanup the gauge metric; deduplication
+	r.addonHealthInfo.Reset()
```

Comment:
Why do we want to do this?

Reply:
This updates the metric with new labels & timestamp and deletes the existing gauge stored in TSDB. 
As it is undesirable to have multiple metrics of the same kind when the state change this ensures `addon_health_info`  has a single time-series per addon.

Reply:
Makes sense, thanks